### PR TITLE
Refactored the code for Show Report

### PR DESCRIPTION
**Pull request by heyrandhir**
[ ] Refactored the code to show the report.

### CODE REVIEW

File: profile/main.go [hunk 1]
```diff
@@ -389,20 +416,20 @@ func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyRespo
...
+		totalProfilesChecked += 1
 		var userId string = doc.Ref.ID
 		var userUrl string
 		var chaincode string
-		
 		if str, ok := doc.Data()["profileURL"].(string); ok {
 			userUrl = str
 		} else {
+			profilesSkipped.ProfileURL = append(profilesSkipped.ProfileURL, userId)
+			logProfileSkipped(client, ctx, userId, "Profile URL not available")
```
Comment (lakshayman) [inline profile/main.go:427]: We can make a global map[string]string for all the constants like status names, skipping reasons, and others.

var Constants map[string]string = map[string]string{
  "constantName": "constantValue",
  ...
}

File: profile/main.go [hunk 2]
```diff
@@ -425,13 +454,68 @@ func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyRespo
...
+			},
+			"ServiceDown": map[string]interface{}{
+				"count":   len(profilesSkipped.ServiceDown),
+				"userIds": profilesSkipped.ServiceDown,
+			},
+			"SameAsLastPendingDiff": map[string]interface{}{
+				"count":   len(profilesSkipped.SameAsLastPendingDiff),
+				"userIds": profilesSkipped.SameAsLastPendingDiff,
+			},
+		},
```
Comment (lakshayman) [inline profile/main.go:508]: Can we make this a function?

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: profile/main.go (hunk 2 above)
Comment (heyrandhir) [reply profile/main.go:539]: yes made a new function to generate report

File: profile/main.go [hunk 3]
```diff
@@ -318,7 +352,8 @@ func generateAndStoreDiff(client *firestore.Client, ctx context.Context, res Res
 /*
  Getting data from the user's service
 */
-func getdata(client *firestore.Client, ctx context.Context, userId string, userUrl string, chaincode string) {
+func getdata(client *firestore.Client, ctx context.Context, userId string, userUrl string, chaincode string) string {
+	var status string = "stored"
```
Comment (lakshayman) [inline profile/main.go:356]: constant left

File: profile/main.go [hunk 4]
```diff
@@ -65,6 +65,20 @@ type Diff struct {
 	Website     string    `firestore:"website,omitempty"`
 }
 
+type Chaincode struct {
```
Comment (lakshayman) [inline profile/main.go:68]: why this structure added?

**Review by lakshayman — CHANGES_REQUESTED** [review]

File: profile/main.go [hunk 5]
```diff
@@ -412,21 +461,83 @@ func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyRespo
 
 		logHealth(client, ctx, userId, isServiceRunning)
 		if !isServiceRunning {
-			setProfileStatusBlocked(client, ctx, userId, "BLOCKED")
+			profilesSkipped.ServiceDown = append(profilesSkipped.ServiceDown, userId)
+			logProfileSkipped(client, ctx, userId, "Service Down")
+			setProfileStatusBlocked(client, ctx, userId, Constants["STATUS_BLOCKED"])
```
Comment (lakshayman) [inline profile/main.go:466]: Here instead of constant we have to give reason behind block which is `Profile Service Down`

File: profile/main.go [hunk 6]
```diff
@@ -385,20 +434,20 @@ func handler(request events.APIGatewayProxyRequest) (events.APIGatewayProxyRespo
...
+		totalProfilesChecked += 1
 		var userId string = doc.Ref.ID
 		var userUrl string
 		var chaincode string
-
 		if str, ok := doc.Data()["profileURL"].(string); ok {
 			userUrl = str
 		} else {
+			profilesSkipped.ProfileURL = append(profilesSkipped.ProfileURL, userId)
+			logProfileSkipped(client, ctx, userId, "Profile URL not available")
```
Comment (lakshayman) [inline profile/main.go:445]: We have to block the service here

File: profile/main.go [hunk 7]
```diff
@@ -332,12 +364,19 @@ func getdata(client *firestore.Client, ctx context.Context, userId string, userU
 	if err != nil {
 		log.Fatal(err)
 	}
-	defer resp.Body.Close()
-
 	if resp.StatusCode == 401 {
-		return
+		status = Constants["PROFILE_SKIPPED_DUE_TO_UNAUTHENTICATED_ACCESS_TO_PROFILE_DATA"]
+		resp.Body.Close()
```
Comment (lakshayman) [inline profile/main.go:369]: Here also we have to log and block service like line no 465

File: profile/main.go [hunk 8]
```diff
@@ -332,12 +364,19 @@ func getdata(client *firestore.Client, ctx context.Context, userId string, userU
...
+		return status
+	}
+	if resp.StatusCode != 200 {
+		status = Constants["PROFILE_SKIPPED_DUE_TO_ERROR_IN_GETTING_PROFILE_DATA"]
+		resp.Body.Close()
```
Comment (lakshayman) [inline profile/main.go:374]: Here also we have to log and block service like line no 465

**Review by lakshayman — CHANGES_REQUESTED** [review]

**Review by lakshayman — APPROVED** [review]

**Pull request merged**